src/spec-lang/tc/typecheck.ts

```diff
@@ -1788,14 +1628,14 @@ export function tcFunctionCall(expr: SFunctionCall, ctx: STypingCtx, typeEnv: Ty
 
         if (def instanceof FunctionDefinition) {
             // param.vType is defined, as you can't put a `var x,` in a function definition.
-            const retTs = def.vReturnParameters.vParameters.map((param) => astVarToSType(param));
+            const retTs = def.vReturnParameters.vParameters.map((param) => astVarToTypeNode(param));
```

Comment:
Can just pass `astVarToTypeNode` instead of creating another closure `(param) => astVarToTypeNode(param)` ?

Reply:
That unfortunately doesn't work due to the extra optional parameters for `astVarTyoTypeNode`. They conflict with the optional parameters for `map` closures.